grovedb/src/replication/state_sync_session.rs

```diff
@@ -342,78 +350,134 @@ impl<'db> MultiStateSyncSession<'db> {
             ));
         }
 
-        let mut next_chunk_ids = vec![];
-
-        let (chunk_prefix, _, _, chunk_id) =
-            replication::utils::decode_global_chunk_id(global_chunk_id, &self.app_hash)?;
+        let mut nested_global_chunk_ids: Vec<Vec<u8>> = vec![];
+        let mut nested_global_chunks: Vec<Vec<u8>> = vec![];
+        if self.app_hash == packed_global_chunk_ids {
+            nested_global_chunk_ids = vec![packed_global_chunk_ids.to_vec()];
+            nested_global_chunks = unpack_nested_bytes(packed_global_chunks)?;
+        } else {
+            nested_global_chunk_ids.extend(unpack_nested_bytes(packed_global_chunk_ids)?);
+            nested_global_chunks.extend(unpack_nested_bytes(packed_global_chunks)?);
+        }
 
+        if (nested_global_chunk_ids.len() != nested_global_chunks.len()) {
+            return Err(Error::InternalError(
+                "Packed num of global chunkIDs and chunks are not matching".to_string(),
+            ));
+        }
         if self.is_empty() {
             return Err(Error::InternalError(
                 "GroveDB is not in syncing mode".to_string(),
             ));
         }
 
-        let current_prefixes = self.as_mut().current_prefixes();
-        let Some(subtree_state_sync) = current_prefixes.get_mut(&chunk_prefix) else {
-            return Err(Error::InternalError(
-                "Unable to process incoming chunk".to_string(),
-            ));
-        };
-        let Ok(res) = subtree_state_sync.apply_inner_chunk(&chunk_id, chunk, grove_version) else {
-            return Err(Error::InternalError("Invalid incoming prefix".to_string()));
-        };
+        let mut next_global_chunk_ids: Vec<Vec<u8>> = vec![];
+
+        for (index, (iter_global_chunk_id, iter_packed_chunks)) in nested_global_chunk_ids
+            .iter()
+            .zip(nested_global_chunks.iter())
+            .enumerate()
+        {
+            let mut next_chunk_ids = vec![];
+
+            let (chunk_prefix, _, _, nested_local_chunk_ids) =
+                replication::utils::decode_global_chunk_id(
+                    iter_global_chunk_id.as_slice(),
+                    &self.app_hash,
+                )?;
+
+            let iter_chunk_ds: Box<dyn Iterator<Item = Vec<u8>>> =
```

Comment:
looks like an overkill and can just return empty vec from true branch and collected vec from false branch